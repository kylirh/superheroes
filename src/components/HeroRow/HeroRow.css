.hide-inactive-heroes .inactive-hero {
    display: none;
}