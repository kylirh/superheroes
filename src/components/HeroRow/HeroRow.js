import React, { Component } from 'react';

class HeroRow extends Component {
  render() {
    const hero = this.props.hero;

    return (
      <tr>
        <td><input type="checkbox" checked={hero.selected} onChange={() => this.props.onSelect(hero)} /></td>
        <td>{hero.id}</td>
        <td>{hero.name}</td>
        <td>{hero.persona}</td>
        <td>{hero.active ? "Active" : "Inactive"}</td>
      </tr>
    );
  }
}

export default HeroRow;
