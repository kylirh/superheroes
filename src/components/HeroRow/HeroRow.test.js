import React from 'react';
import HeroRow from './HeroRow';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe("HeroRow", () => {
    it("renders correctly", () => {
        const hero = { id: 1, name: "Konrad Sanro", persona: "Retwoner", active: false };
        const component = shallow(
            <HeroRow hero={hero} key={hero.id} />
        );
        expect(shallowToJson(component)).toMatchSnapshot();
    });

    it("calls the passed select method", () => {
        const hero = { id: 1, name: "Konrad Sanro", persona: "Retwoner", active: false };
        const handleSelect = jest.fn();
        const component = shallow(
            <HeroRow hero={hero} key={hero.id} onSelect={handleSelect} />
        );
        component.find("input").simulate("change");
        expect(handleSelect).toHaveBeenCalledWith(hero);
    });
})
