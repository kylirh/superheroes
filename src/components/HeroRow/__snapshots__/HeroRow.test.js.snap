// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroRow renders correctly 1`] = `
<tr
  className="hero-row"
>
  <td
    className="cell-center"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
  </td>
  <td
    className="cell-center"
  >
    1
  </td>
  <td>
    Konrad Sanro
  </td>
  <td>
    Retwoner
  </td>
  <td>
    Inactive
  </td>
</tr>
`;
